public/app/features/datasources/components/EditDataSource.tsx

```diff
@@ -106,9 +107,14 @@ export function EditDataSourceView({
   const { readOnly, hasWriteRights, hasDeleteRights } = dataSourceRights;
   const hasDataSource = dataSource.id > 0;
 
+  const ds = getDataSourceSrv()?.getInstanceSettings(dataSource.uid);
```

Comment:
I agree 💯, for some reason `dataSourceMeta.alerting` is undefined. The only way we found getting this alerting was using the `grafanaBootData.settings.datasources.{DATASOURCE}.meta.alerting`
We could create an issue for improving this @konrad147, but I would not block this fix because of that.